docs/designs/update-strategy.md

```diff
@@ -0,0 +1,661 @@
+
+## Architecture
+
+### Three-Level Update Control
+
+```
+PodCliqueSet (Top Level)
+├─ UpdateStrategy (controls PCS replica updates)
+│  ├─ RollingUpdate: one replica at a time
+│  └─ ReplicaRecreate: tear down entire replica
+│
+├─ PCS Replica 0
+│  ├─ Standalone PodCliques (concurrent updates)
+│  │  └─ frontend: UpdateStrategy (controls pod updates)
+│  │     └─ maxUnavailable/maxSurge
+│  │
+│  └─ PodCliqueScalingGroups (concurrent updates)
+│     ├─ prefill: UpdateStrategy (controls PCSG replica updates)
+│     │  └─ maxUnavailable/maxSurge
+│     └─ decode: UpdateStrategy (controls PCSG replica updates)
+│        └─ maxUnavailable/maxSurge
+│
+└─ PCS Replica 1, 2, ... (sequential per PCS strategy)
+```
+
+## API Structure
+
+### 1. PodCliqueSetSpec.UpdateStrategy
+
+Controls **how PCS replicas are updated** (the top-most level).
+
+```go
+type PodCliqueSetUpdateStrategy struct {
+    Type          PodCliqueSetUpdateStrategyType
+    RollingUpdate *PodCliqueSetRollingUpdateStrategy
+}
+
+type PodCliqueSetUpdateStrategyType string
+const (
+    RollingUpdate    // Update replicas sequentially
+    ReplicaRecreate  // Delete/recreate entire replica
+)
+
+type PodCliqueSetRollingUpdateStrategy struct {
+    MaxUnavailable *intstr.IntOrString  // How many PCS replicas can be down (absolute number or percentage)
+    MaxSurge       *intstr.IntOrString  // How many extra PCS replicas to create (absolute number or percentage)
+}
+```
+
+**Defaults:** `RollingUpdate` with `maxUnavailable=1, maxSurge=0`
+
+### 2. PodCliqueTemplateSpec.UpdateStrategy
+
+Controls **how pods update within a standalone PodClique**.
+
+```go
+type ComponentUpdateStrategy struct {
+    MaxUnavailable *intstr.IntOrString  // How many pods can be down (absolute number or percentage)
+    MaxSurge       *intstr.IntOrString  // How many extra pods to create (absolute number or percentage)
+}
+```
+
+**Defaults:** `maxUnavailable=1, maxSurge=0`
+**Scope:** Only applies to standalone PodCliques (not in a PCSG)
+
+### 3. PodCliqueScalingGroupConfig.UpdateStrategy
+
+Controls **how PCSG replicas update within a scaling group**.
+
+```go
+type ComponentUpdateStrategy struct {
+    MaxUnavailable *intstr.IntOrString  // How many PCSG replicas can be down (absolute number or percentage)
+    MaxSurge       *intstr.IntOrString  // How many extra PCSG replicas to create (absolute number or percentage)
+}
+```
+
+**Defaults:** `maxUnavailable=1, maxSurge=0`
+**Scope:** Controls the update of PCSG replicas (groups of PodCliques)
+
+## Update Behavior
+
+### RollingUpdate (Default)
```

Comment:
ReplicaUpdate

Reply:
Actually another way to Frame it DelegateChildren or something like that. IE follow the strategy of the children vs replica recreate which ignores their strategies and just recreates.